### DISCUSSION TITLE
Is it possible to add templates to a note that is NOT the workspace note?

### DISCUSSION THREAD

**Discussion by Fontanapink** [windingwind/zotero-better-notes#817] [Q&A] [answered]
I am trying to reference one work on an item's note by using [this](https://github.com/windingwind/zotero-better-notes/discussions/569) but it won't work.

**Comment by windingwind** [comment, marked as answer]
https://github.com/windingwind/zotero-better-notes#note-editor

See the image: Click the top-right button and you'll find a menu `Insert template to cursor line`

**Reply by Fontanapink** [reply]
I tried. But this works for the "workspace note" and not an item's note right? At least it is not working for me.

Thank you for the great work!

**Reply by windingwind** [reply]
For every note. The `Link current note to workspace` is only for workspace note; but the one I showed to you is not, at least this is the case with the latest beta version.

**Reply by Calorion** [reply]
Where is `Insert template to cursor line`? I only see `Insert template`, which inserts the template at the top of the note, not at the insertion point.

**Reply by windingwind** [reply]
You can insert a link to another note into the cursor of the current note:

<img width="1134" alt="image" src="https://github.com/user-attachments/assets/b5abfbb0-1cef-4ad9-84f6-086f6cef8232" />


However, the link itself does not contain any line/section information. If you want to link to a specific line/section of a note, use 

<img width="355" alt="image" src="https://github.com/user-attachments/assets/8b73db0a-c975-48c8-a029-0ae89abe3f60" />

**Reply by Calorion** [reply]
But I’m trying to add a template to a note, such as the “Current Time” template in the “[About Note Templates](https://github.com/windingwind/zotero-better-notes/blob/master/docs/about-note-template.md)” doc.

**Reply by windingwind** [reply]
Template is always inserted into the cursor line

**Reply by Calorion** [reply]
But it’s not. It’s showing up at the top of the note, not the insertion point (which I assume is what you mean by “cursor line”). For the Current Time template, I just want the time inserted right where I’m typing. Or am I misunderstanding something?

**Reply by windingwind** [reply]
Sorry but I don't understand what do you mean by `insertion point `. Do you have the cursor active in the note editor?

**Reply by Calorion** [reply]
I think what you’re calling a “cursor” is actually an insertion point (cursors only exist in terminals). It’s the blinky line where you enter text. 

And yes, it’s active in the note editor when I invoke the template.

…except now it’s working. I think it’s the same bug as #1239.

**Reply by windingwind** [reply]
OK, if you confirm it's fixed, I can push it to the stable release channel.

**Reply by Calorion** [reply]
No. I wasn’t using the beta previously; it just behaves the same as the other issue: When invoked on a blank line, it works; when invoked anywhere else, it appears above the section containing the insertion point.

#1239 is fixed in the beta, however.
//
Or…er…sort of? Now it’s sometimes dropping references to the very bottom of the note even when a section is selected, which it wasn’t doing before. I’ll see whether I can isolate it.
